### PR TITLE
Revert "Replace `ORDER BY ALL` by `ORDER BY *`"

### PR DESCRIPTION
**Pull request by rschu1ze**
#57875 introduced syntax `ORDER BY ALL`. 

Turned out that there are ambiguities when column "all" already exists in the table or in the `SELECT` clause. For example, for a table with columns `all`, `a`,`b`:

```sql
SELECT all, a, b ORDER BY all;        -- should we sort by all columns in SELECT or by column "all"?
SELECT a, b AS all ORDER BY all;      -- like before but "all" as alias
SELECT a, func(...) AS all ORDER BY all; -- like before but "all" as function
SELECT a, b ORDER BY all;             -- tricky in other way: does the user want to sort by all columns in SELECT clause or by not SELECTed column "all"? (<-- likely the latter)
```

After some discussion in #59450, `ORDER BY ALL` was replaced by `ORDER BY *` which avoids all ambiguities as `*` cannot be used as an identifier (e.g. for column names).

Haha, there is a new ambiguity:

```sql
SELECT a, b ORDER BY *;             -- sort by all columns in SELECT clause or by all columns in table (analogous to SELECT *)?
```

Also, I heard that `ORDER BY ALL` is already established syntax.

So this PR does:
- https://github.com/ClickHouse/ClickHouse/pull/60248/commits/52afa46e237fb16705574170c94999527c6b5ddf Roll back #59450
- https://github.com/ClickHouse/ClickHouse/pull/60248/commits/7930c2a993521a18558307ea19b6df9acc49fd5f Make some fixes to improve the previous handling of ambiguity

The previous handling of ambiguity is to throw an exception. This happens when 1. `all` is a column name in the table or part of the `SELECT` clause and 2. `ORDER BY ALL` is used. The exception suggests to disable `ORDER BY ALL` to make the query succeed.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Restore the previous syntax `ORDER BY ALL` which has temporarily (for a few days) been replaced by ORDER BY *.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 52afa46e237fb16705574170c94999527c6b5ddf with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60248/52afa46e237fb16705574170c94999527c6b5ddf/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60248/52afa46e237fb16705574170c94999527c6b5ddf/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: docs/en/operations/settings/settings.md [hunk 1]
```diff
@@ -4279,6 +4279,41 @@ Result:
...
+Query:
+
+```sql
+CREATE TABLE TAB(C1 Int, C2 Int, ALL Int) ENGINE=Memory();
+
+INSERT INTO TAB VALUES (10, 20, 30), (20, 20, 10), (30, 10, 20);
+
+SELECT * FROM TAB ORDER BY ALL; -- returns an error that ALL is ambiguous
+
+SELECT * FROM TAB ORDER BY ALL SETTINGS enable_order_by_all;
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply docs/en/operations/settings/settings.md:4304]: I am not done here yet, new commit is coming

[the remark this answers is not in the document]
Comment (rschu1ze) [reply docs/en/operations/settings/settings.md:4304]: And yes, it missed a `= 0`.

**Review by alexey-milovidov — APPROVED** [review]

File: src/Core/Settings.h [hunk 2]
```diff
@@ -872,6 +872,7 @@ class IColumn;
     M(UInt64, cache_warmer_threads, 4, "Only available in ClickHouse Cloud", 0) \
     M(Int64, ignore_cold_parts_seconds, 0, "Only available in ClickHouse Cloud", 0) \
     M(Int64, prefer_warmed_unmerged_parts_seconds, 0, "Only available in ClickHouse Cloud", 0) \
+    M(Bool, enable_order_by_all, true, "Enable sorting expression ORDER BY ALL.", 0) \
```
Comment (Algunenano) [inline src/Core/Settings.h:875]: Is this setting in the experimental group by mistake or is it really experimental?

Comment (rschu1ze) [reply src/Core/Settings.h:875]: By mistake (though a few settings in the section don't look experimental).

Fix: https://github.com/ClickHouse/ClickHouse/pull/60449

**Pull request merged**